src/docs/asciidoc/modules/ROOT/pages/chapter4.adoc

```diff
@@ -142,394 +214,537 @@ entity Tag {
-
-You'll need a service-discovery server installed before you can start the gateway. You'll also need Keycloak for authentication and MongoDB for your store microservices. The blog application depends on Elasticsearch and PostgreSQL, but only when running in production mode. Luckily, JHipster creates Docker Compose files for all of the services your apps depend on. You can run the following commands from your project's root directory to start Docker containers for JHipster Registry, Keycloak, and MongoDB.
+Start the `blog` by opening a terminal and navigating to its directory. Then, start its database with Docker and the app with Gradle.
 
+[source,shell]
 ----
-docker-compose -f gateway/src/main/docker/jhipster-registry.yml up -d
-docker-compose -f gateway/src/main/docker/keycloak.yml up -d
-docker-compose -f store/src/main/docker/mongodb.yml up -d
+npm run docker:db:up
+./gradlew
 ----
 
-By default, the JHipster Registry will use Keycloak when running with Docker Compose. For everything to work properly, you'll need to add an entry to your `hosts` file for Keycloak.
+Open a new terminal and do the same for the `store` microservice.
 
-----
-127.0.0.1       keycloak
-----
+You can verify everything is started using Consul at `\http://localhost:8500`.
 
-If you want to change it to use Okta, you'll need to modify `gateway/src/main/docker/jhipster-registry.yml` and change the default Keycloak settings to use your Okta settings, or environment variables (recommended).
+.Consul services
+image::chapter4/consul-services.png[Consul services, 1184, scaledwidth=100%, align=center]
 
-[source%autofit,yaml]
-----
-- SECURITY_OAUTH2_CLIENT_ACCESS_TOKEN_URI=${SECURITY_OAUTH2_CLIENT_ACCESS_TOKEN_URI}
-- SECURITY_OAUTH2_CLIENT_USER_AUTHORIZATION_URI=${SECURITY_OAUTH2_CLIENT_USER_AUTHORIZATION_URI}
-- SECURITY_OAUTH2_CLIENT_CLIENT_ID=${SECURITY_OAUTH2_CLIENT_CLIENT_ID}
-- SECURITY_OAUTH2_CLIENT_CLIENT_SECRET=${SECURITY_OAUTH2_CLIENT_CLIENT_SECRET}
-- SECURITY_OAUTH2_RESOURCE_USER_INFO_URI=${SECURITY_OAUTH2_RESOURCE_USER_INFO_URI}
-----
+Refresh the gateway app; you should see menu items to navigate to the microservices now.
 
-TIP: You can also put these variables in a file and specify an `env_file` setting. See https://docs.docker.com/compose/environment-variables/#the-env-file[Environment variables in Compose] to learn more.
+.Gateway entities available
+image::chapter4/gateway-entities-available.png[Consul services, 1132, scaledwidth=100%, align=center]
 
-Then you'll need to restart your JHipster Registry (if it's already running):
-
-----
-docker-compose -f gateway/src/main/docker/jhipster-registry.yml restart
-----
+=== Zero turnaround development that sparks joy
 
-To log in, you'll need to add `http://localhost:8761/login` as a login-redirect URI in your Okta app.
+At this point, I've only shown you how to run the Spring Boot backends with their packaged React apps. What if you want to work on the UI and have zero turnaround that sparks joy? ✨🤗
```

Comment:
>with their packaged React apps
Do you mean to say React apps or reactive apps? 
I'm confused here since you're specifying Angular and mention vue, but not exactly React.

Reply:
React is used in this example. You can see the JDL for it at https://github.com/jhipster/jdl-samples/blob/main/reactive-mf.jdl. In the beginning, there's this paragraph:

> This tutorial shows you how to build a microservices architecture with JHipster 7.9.3. You’ll generate a gateway (powered by Spring Cloud Gateway), a blog microservice (that talks to Neo4j), a store microservice (that uses MongoDB), and use Docker Compose to make sure it all runs locally. I’ll also provide some pointers on how to deploy it with Kubernetes.

And I mention React is the framework in the JDL. Maybe I should change the opening paragraph to be something like this:

> ... Docker Compose to make sure it all runs locally. The gateway will contain a React shell app that loads the blog and store micro frontends. I’ll also provide some pointers ...

Reply:
I could also say "run the Spring Boot backends with their packaged React micro frontends."

Reply:
I like the specificity of that sentence 


```suggestion
At this point, I've only shown you how to run the Spring Boot backends with their packaged React micro frontends. What if you want to work on the UI and have zero turnaround that sparks joy? ✨🤗
```

Reply:
Fixed in https://github.com/mraible/jhipster-book/pull/76/commits/3c63c78b317cdc8ac42157a6f1b8df06fcd22d2e.